### PR TITLE
Feature arm fp16 op

### PR DESCRIPTION
**Pull request by powerpwang**
1. Support ARMv8.2 FP16
2. FP16 OP: Add, BatchNorm, Concat, Conv, Deconv, Pool, Permute,  PReLU, ReLU, ReLU6, Sigmoid, Softmax

### CODE REVIEW

File: test/unit_test/layer_test/test_conv_layer.cc [hunk 1]
```diff
@@ -25,26 +25,27 @@ class ConvLayerTest : public LayerTest,
...
+                             testing::Values(1, 2, 3, 5),
+                             // dilation
+                             testing::Values(1, 2),
+                             // stride
+                             testing::Values(1, 2),
+                             // pads
+                             testing::Values(0, 1),
+                             // data_type
+                             testing::Values(DATA_TYPE_FLOAT),
+                             // activation_type
```
Comment (seanxcwang) [inline test/unit_test/layer_test/test_conv_layer.cc]: merge完master之后，half的测试没了

**Review by seanxcwang — DISMISSED** [review]

**Review by seanxcwang — APPROVED** [review]
